compiler/rustc_const_eval/src/interpret/eval_context.rs

```diff
@@ -114,7 +114,6 @@ impl<'tcx, M: Machine<'tcx>> FnAbiOfHelpers<'tcx> for InterpCx<'tcx, M> {
 }
 
 /// Test if it is valid for a MIR assignment to assign `src`-typed place to `dest`-typed value.
-/// This test should be symmetric, as it is primarily about layout compatibility.
 pub(super) fn mir_assign_valid_types<'tcx>(
```

Comment:
we check for subtyping here which is *not symmetric*, so either this comment is wrong or the impl is. I would be open to alternative change this to try subtyping in either direction :shrug: but given that this already uses `src` and `dest` and the behavior is preexisting, this comment is not relied on rn

Reply:
I think I originally made the check symmetric since I didn't ant to think about which direction it has to be. :joy: So it might be worth checking that all users of this function use it correctly.

I also find the comment extremely confusing. When doing an assignment, `dest` is a place and `src` is a value! So what is going on here?
What I expect is a comment like this:
```
/// Test if it is valid for a MIR assignment to store a `src`-typed value in a `dest`-typed place.
```
But I am not sure if that matches the implementation, or the users...

Reply:
Apparently I [wrote](https://github.com/rust-lang/rust/commit/f1ea2731335b1e6c84ebd2d4cd767626ee1073df) that comment. Unfortunately I can't explain why...